### PR TITLE
Add sample application update_test.

### PR DESCRIPTION
**Pull request by btheosam**
Add update_test sample application to test
OS update functionality.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3b17d416b09585fd17ac16574fdfd7a377adc990] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [3b17d416b09585fd17ac16574fdfd7a377adc990] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [eedfd07a19573be1f56b0f7bf5c37119c521e14c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [eedfd07a19573be1f56b0f7bf5c37119c521e14c] - Code Rule Check OK.

File: apps/examples/update_test/Make.defs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+############################################################################
+# apps/examples/hello/Make.defs
```
Comment (NilabhShant) [inline apps/examples/update_test/Make.defs]: Please correct the path.

File: apps/examples/update_test/update_test_main.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * examples/hello/hello_main.c
```
Comment (NilabhShant) [inline apps/examples/update_test/update_test_main.c]: same as above.

File: apps/examples/update_test/Make.defs (hunk 1 above)
Comment (juitem) [inline apps/examples/update_test/Make.defs]: apps/examples/update_test/Make.defs

File: build/configs/artik053/update_test/Make.defs [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+############################################################################
+# configs/artik053/update_test/Make.defs
```
Comment (NilabhShant) [inline build/configs/artik053/update_test/Make.defs]: Please make: configs/artik053/update_test/Make.defs  --> build/configs/artik053/update_test/Make.defs

File: apps/examples/update_test/update_test_main.c (hunk 2 above)
Comment (juitem) [inline apps/examples/update_test/update_test_main.c]: examples/update_test/update_test_main.c

**Comment by seinfra** [comment]
Target : [41cc318a78d44bb7570bda89bf8f1fb9a2f5bfce] - Code Rule Check OK.

File: apps/examples/update_test/Make.defs (hunk 1 above)
Comment (btheosam) [reply apps/examples/update_test/Make.defs]: Thanks, fixed.

File: apps/examples/update_test/update_test_main.c (hunk 2 above)
Comment (btheosam) [reply apps/examples/update_test/update_test_main.c]: Thanks, fixed.

File: apps/examples/update_test/Make.defs (hunk 1 above)
Comment (btheosam) [reply apps/examples/update_test/Make.defs]: Thanks, fixed.

File: apps/examples/update_test/update_test_main.c (hunk 2 above)
Comment (btheosam) [reply apps/examples/update_test/update_test_main.c]: Thanks, fixed.

File: build/configs/artik053/update_test/Make.defs (hunk 3 above)
Comment (btheosam) [reply build/configs/artik053/update_test/Make.defs]: Thanks, fixed.

**Comment by seinfra** [comment]
Target : [41cc318a78d44bb7570bda89bf8f1fb9a2f5bfce] -

**Comment by seinfra** [comment]
Target : [b9f8c5b01fa9b98a3b5031582ba2e8bf78cc6d56] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b9f8c5b01fa9b98a3b5031582ba2e8bf78cc6d56] - Code Rule Check OK.

File: apps/examples/update_test/update_test_main.c [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * examples/update_test/update_test_main.c
```
Comment (NilabhShant) [inline apps/examples/update_test/update_test_main.c]: Please make it app/examples/update_test/update_test_main.c

Comment (btheosam) [reply apps/examples/update_test/update_test_main.c]: Thanks, fixed.

**Comment by seinfra** [comment]
Target : [4c5d722ddec1602d15af470ea4e2aa4941ed7d7e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4c5d722ddec1602d15af470ea4e2aa4941ed7d7e] - Code Rule Check OK.

**Pull request merged**